File: LeetCode/Simulation/JS/Leetcode1380-矩阵中的幸运数.js
var luckyNumbers  = function(matrix) {
    const n = matrix.length,m = matrix[0].length;
    const row = new Array(n).fill(100000),col = new Array(m).fill(0);
    for(let i = 0;i < n;i++)
    {
        for(let j = 0;j < m;j++)
        {
            row[i] = Math.min(row[i],matrix[i][j]);
            col[j] = Math.max(col[j],matrix[i][j]);
        }
    }
    const ans = [];
    for(let i = 0;i < n;i++)
    {
        for(let j = 0;j < m;j++)
        {
            if(matrix[i][j] == col[j] && matrix[i][j] == row[i]) ans.push(col[j]);
        }
    }
    return ans;
};

File: LeetCode/Simulation/Rust/Leetcode1380-矩阵中的幸运数.rs
impl Solution {
  pub fn lucky_numbers(matrix: Vec<Vec<i32>>) -> Vec<i32> {
    let (n, m) = (matrix.len(), matrix[0].len());
    let (mut row, mut col) = (vec![100005; n], vec![0; m]);
    for i in 0..n {
      for j in 0..m {
        row[i] = row[i].min(matrix[i][j]);
        col[j] = col[j].max(matrix[i][j]);
      }
    }
    let mut ans = Vec::new();
    for i in 0..n {
      for j in 0..m {
        if row[i] == matrix[i][j] && matrix[i][j] == col[j] {
          ans.push(row[i]);
        }
      }
    }
    ans
  }
}
